### PR TITLE
Revise setup script

### LINKED ISSUE #145: Update setup scripts (especially for setting up postgres)
The current [`scripts/`](https://github.com/gcivil-nyu-org/spring2020-cs-gy-9223-class/commits/70e3d10d35df7fba5262b03a45b82153fa94b93f/scripts) folder contains two scripts for checking and setting up workspace. However, they are outdated, and cause problems for both new end users and developers.

## Expected Behavior
The scripts work as they are described in `README.md`.

## Current Behavior
The script do not fully work, and can mess up user's or developer's workspace configurations.

## Ideas for Improvement
Update commands in these files to make sure they work for all develop teams and end users.
Also update `settings.py` and other affected files.

### PR DESCRIPTION
**Pull request by alldne**
Closes #145 

Please test this script in a fresh environment. Leave a comment and let me know if it worked with your environment (especially OS).

**_Do not merge this PR until it's tested with major OS i.e Windows, linux (including ubuntu), macOS at least once._**

# 1. Newly introduced environment variables in `.env` file
For instance,
```
DB=postgres
DB_USER=mercury
DB_PASSWORD=mercury
DB_HOST=1.2.3.4
DB_PORT=12345
```
You can omit any of those variables other than `DB=postgres`. When it's not given, default values are applied.

For backward compatibility and heroku, it still supports `DATABASE_URL`. That means, this is still valid `.env` file.
```
DATABASE_URL=postgres://postgres@localhost:5432/mercury
```

# 2. Revised scripts/setup.sh:
- Checking dependencies to be installed:
    - postgres
    - geckodriver
- Creating mercury database
- Add missing `python manage.py collectstatic`
- Getting more interactive

### CODE REVIEW

**Comment by dandipietrantonio** [comment]
Works on Linux Mint, still waiting to here about Windows and Mac before approving.

**Comment by alldne** [comment]
Works on macOS Catalina 10.15.3

Originally I was merging it into our team branch first but it's okay to merge it directly to master.

**Comment by PrabhanshuAttri** [comment]
Are we providing support on windows, as the setup script is a bash script? If yes, could you please also write a small write up to do that.

**Comment by iamvibhorsingh** [comment]
Works on windows 10!

**Comment by alldne** [comment]
@PrabhanshuAttri Yes, we support Windows and I think we should do. Personally I try not to give any assumption on the end-user's environment. Eventually a motor sports team member will follow through these steps too.

By the way, I'm not sure I understand you correctly. I have no idea how bash script is related with Windows and what implication is behind it.  Could you rephrase your sentence again?

**Comment by PrabhanshuAttri** [comment]
I was referring to `setup.sh` file. I was not sure how to run this file on the windows. As @iamvibhorsingh has tested it, I think we are good to go.

**Comment by PrabhanshuAttri** [comment]
@alldne I noticed the coveralls have reduced by 0.2%. Can we check why is it so?

**Comment by alldne** [comment]
> @alldne I noticed the coveralls have reduced by 0.2%. Can we check why is it so?

Let me take a look later today!

**Comment by alldne** [comment]
@PrabhanshuAttri The coverage drop was because I updated lines in `mysite/settings.py` where database configuration is made but I omitted `#pragma no cover`.

These lines refer to machine's environment variables to configure database. And some code is never executed depending on the configuration. Now `#pragma no cover` has been added.
